src/formatDate.js

```diff
@@ -50,7 +50,58 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fmFormatSeparator = fromFormat[fromFormat.length - 1];
+  const toFormatSeparator = toFormat[toFormat.length - 1];
+  const dateArray = date.split(fmFormatSeparator);
+
+  const changedDate = [];
+
+  let year, month, day;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      fromFormat[i] = dateArray[i];
+      year = fromFormat[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      fromFormat[i] = dateArray[i];
+      month = fromFormat[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      fromFormat[i] = dateArray[i];
+      day = fromFormat[i];
+    }
+  }
+
+  const makeYear = () => {
+    if (year < 30) {
+      return changedDate.push(20 + year);
+    }
+
+    return changedDate.push(19 + year);
+  };
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'YY' && year.length === 4) {
+      changedDate.push(year.slice(-2));
+    } else if (toFormat[i] === 'YYYY' && year.length === 2) {
+      makeYear(year);
```

Comment:
You use function only once, so maybe it's better just write a function body right here

Reply:
Done